### PR TITLE
Untrusted RLP length overflow check 

### PR DESCRIPTION
**Pull request by NikVolf**

### CODE REVIEW

File: util/rlp/src/untrusted_rlp.rs [hunk 1]
```diff
@@ -371,7 +371,8 @@ impl<'a> BasicDecoder<'a> {
 				}
 				let len = decode_usize(&bytes[1..begin_of_value])?;
 
-				let last_index_of_value = begin_of_value + len;
+				let last_index_of_value = begin_of_value.overflowing_add(len)
```
Comment (rphmeier) [inline util/rlp/src/untrusted_rlp.rs]: does this compile? i thought `overflowing_add` returns `(usize, bool)`. `checked_add` returns an `Option`, though.

File: util/rlp/src/untrusted_rlp.rs [hunk 2]
```diff
@@ -385,7 +386,7 @@ impl<'a> BasicDecoder<'a> {
 
 #[cfg(test)]
 mod tests {
-	use UntrustedRlp;
+	use {UntrustedRlp, DecodeError};
```
Comment (rphmeier) [inline util/rlp/src/untrusted_rlp.rs]: `DecoderError`

File: util/rlp/src/untrusted_rlp.rs (hunk 1 above)
Comment (NikVolf) [inline util/rlp/src/untrusted_rlp.rs]: i'm not putting pleasereview for a reason here :)
i have some issues with cargo here

**Comment by NikVolf** [comment]
Fixes #6226

**Comment by rphmeier** [comment]
LGTM

**Pull request merged**